cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java

```diff
@@ -0,0 +1,344 @@
+        Optional<List<PackageDescriptor.Dependency>> baloWithAllDependenciesAvailableInCentral = balos.values().stream()
+                .filter(List::isEmpty)
+                .findAny();
+
+        // if there isn't any balos where dependencies are resolved, then throw an error.
+        if (!baloWithAllDependenciesAvailableInCentral.isPresent()) {
+            Set<String> unresolvedDependencies = balos.values().stream()
+                    .flatMap(List::stream)
+                    .map(PackageDescriptor.Dependency::toString)
+                    .collect(Collectors.toSet());
+            throw createLauncherException("unable to find dependencies in remote repository: [" +
+                    String.join(", ", unresolvedDependencies) + "]");
+        }
+
+        // push all the modules where dependencies are available in remote repository and remove them from the map.
+        Iterator<Map.Entry<Path, List<PackageDescriptor.Dependency>>> iterator = balos.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<Path, List<PackageDescriptor.Dependency>> baloDeps = iterator.next();
+            if (baloDeps.getValue().isEmpty()) {
+                pushBaloToRemote(baloDeps.getKey());
+                iterator.remove();
+            }
+        }
+        recursivelyPushBalos(balos);
+    }
+
+    /**
+     * Push a balo file to remote repository.
+     *
+     * @param baloPath Path to the balo file.
+     */
+    private static void pushBaloToRemote(Path baloPath) {
+        Path baloFileName = baloPath.getFileName();
+        if (null != baloFileName) {
+            // Load BaloProject from balo path
+            BaloProject baloProject = BaloProject.loadProject(baloPath, null);
+            String name = baloProject.currentPackage().packageName().toString();
+
+            try {
+                CentralAPIClient client = new CentralAPIClient();
+                client.pushPackage(baloPath, baloProject);
+            } catch (CommandException e) {
+                String errorMessage = e.getMessage();
+                if (null != errorMessage && !"".equals(errorMessage.trim())) {
+                    // removing the error stack
+                    if (errorMessage.contains("\n\tat")) {
+                        errorMessage = errorMessage.substring(0, errorMessage.indexOf("\n\tat"));
+                    }
+
+                    errorMessage = errorMessage.replaceAll("error: ", "");
+
+                    throw createLauncherException(
+                            "unexpected error occurred while pushing package '" + name + "' to remote repository("
+                                    + getRemoteRepoURL() + "): " + errorMessage);
+                }
+            }
+        }
+    }
+
+    private static boolean isDependencyAvailableInRemote(PackageDescriptor.Dependency dep) {
+        List<String> supportedPlatforms = Arrays.stream(SUPPORTED_PLATFORMS).collect(Collectors.toList());
```

Comment:
The dependencies has to be resolved from project API since some dependencies might not be in descriptor in some cases.

Reply:
This checks whether the package already available in the central, renamed method to `isPackageAvailableInRemote`